vsdx/__init__.py

```diff
@@ -383,10 +383,11 @@ def __init__(self, xml: Element, parent_xml: Element, page: VisioFile.Page):
         def __repr__(self):
             return f"<Shape tag={self.tag} ID={self.ID} type={self.type} text='{self.text}' >"
 
-        def copy(self):
+        def copy(self, page: VisioFile.Page = None) -> VisioFile.Shape:
+            page = page or self.page
             self.page.set_max_ids()
-            new_shape_xml = self.page.vis.copy_shape(self.xml, self.page.xml, self.page.filename)
-            return VisioFile.Shape(xml=new_shape_xml, parent_xml=self.parent_xml, page=self.page)
+            new_shape_xml = self.page.vis.copy_shape(self.xml, page.xml, page.filename)
+            return VisioFile.Shape(xml=new_shape_xml, parent_xml=page.xml, page=page)
```

Comment:
Hi - note that parent_xml arg here should be the parent shape or shapes element, not the page.xml (so in this case if page is passed parent_xml would be first Shapes tag in that destination page)

Reply:
Got it, will be sure to amend the update with this in mind.